src/Persistence/Templates/ControlFactory.cs

```diff
@@ -15,21 +15,21 @@ public ControlFactory(IControlTemplateStore controlTemplateStore)
         _controlTemplateStore = controlTemplateStore ?? throw new ArgumentNullException(nameof(controlTemplateStore));
     }
 
-    public Control Create(string name, string template, ControlPropertiesCollection? properties = null, Control[]? children = null)
+    public Control Create(string name, string template, ControlPropertiesCollection? properties = null, IList<Control>? children = null)
     {
         if (_controlTemplateStore.TryGetTemplateByName(template, out var controlTemplate))
         {
             return new BuiltInControl(name, controlTemplate)
             {
                 Properties = properties ?? new(),
-                Children = children ?? Array.Empty<Control>()
+                Children = children
```

Comment:
please add a fallback when null

Reply:
it can be null. Fallback from null to null?

Reply:
that's the point, it should be empty collection

Reply:
Added "This collection can be null in cases where the control does not support children" to the property